lib/src/base.dart

```diff
@@ -753,47 +753,89 @@ class Device {
         break;
     }
 
-    if (name != null) json['name'] = name;
+    if (name != null) {
```

Comment:
These if null ... look like good candidates for collection if
Not for this PR, but I think it would make the code look nicer